relayer/tests/browser_test.ts

```diff
@@ -0,0 +1,152 @@
+//@ts-check
+import { AnchorProvider, BN } from "@coral-xyz/anchor";
+import {
+  Connection,
+  Keypair as SolanaKeypair,
+  PublicKey,
+} from "@solana/web3.js";
+import chai, { expect } from "chai";
+import chaiHttp from "chai-http";
+import express from "express";
+import { sign } from "tweetnacl";
+
+import {
+  Provider,
+  airdropSol,
+  User,
+  Relayer,
+  useWallet,
+  ADMIN_AUTH_KEYPAIR,
+  createTestAccounts,
+} from "@lightprotocol/zk.js";
+import sinon from "sinon";
+
+import {
+  updateMerkleTree,
+  getIndexedTransactions,
+  handleRelayRequest,
+  buildMerkleTree,
+  getLookUpTable,
+} from "../src/services";
+
+chai.use(chaiHttp);
+const app = express();
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+// Use sinon to create a stub for the middleware
+const addCorsHeadersStub = sinon
+  .stub()
+  .callsFake((_req: any, _res: any, next: any) => next());
+app.use(addCorsHeadersStub);
+
+app.post("/updatemerkletree", updateMerkleTree);
+app.get("/lookuptable", getLookUpTable);
+app.post("/relayTransaction", handleRelayRequest);
+app.get("/indexedTransactions", getIndexedTransactions);
+app.get("/getBuiltMerkletree", buildMerkleTree);
+
+describe("Browser tests", () => {
+  var RELAYER: Relayer;
+  const walletMock = useWallet(ADMIN_AUTH_KEYPAIR);
+  const connection = new Connection("http://127.0.0.1:8899", "confirmed");
+
+  before(async () => {
```

Comment:
Can you explain how these tests replicate a browser env?

Reply:
runScript.sh runs it after initing browser env with jsdom; 
npx mocha -r ts-node/register -r jsdom-global/register -r ./setup.jsdom.ts tests/browser_test.ts --timeout 1000000 --exit;

browser_test.ts has 2 tests: 1) executing the core functionality (can expand in future) for success and 2) for failure (using a node feature like anchorprovider in browser env)

Reply:
ok so jsdom is initing a browser env and all tests after are executed in this env

Reply:
yes